src/moonlink/src/storage/iceberg/iceberg_table_config.rs

```diff
@@ -4,17 +4,30 @@ use std::collections::HashMap;
 
 #[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
 pub struct RestCatalogConfig {
+    #[serde(rename = "name")]
+    #[serde(default)]
+    pub name: String,
+
     #[serde(rename = "uri")]
     #[serde(default)]
-    uri: String,
+    pub uri: String,
 
     #[serde(rename = "warehouse")]
     #[serde(default)]
-    warehouse: String,
-
+    pub warehouse: String,
+
+    /// Optional configuration properties:
+    ///
+    /// - prefix:          Optional URL path prefix to insert after the base URI and API version.
+    /// - oauth2-server-uri: Custom OAuth2 server URI. Defaults to: [uri, PATH_V1:"v1", "oauth", "tokens"].join("/")
+    /// - token:           Static authentication token used by the client for sending requests.
+    /// - credentials:     Client credentials used to fetch a new token.
+    ///     - None: No credentials provided.
+    ///     - Some(None, client_secret): Only client_secret is provided.
+    ///     - Some(Some(client_id), client_secret): Both client_id and client_secret are provided.
```

Comment:
My personal preference is, if there're not too many fields, I would put all of them as field member.
But I don't have a strong opinion here, as long as we have documentation.

nit: worth calling out other unknown fields will be ignored.